insights/combiners/crio_conf.py

```diff
@@ -0,0 +1,150 @@
+
+from insights.contrib.ConfigParser import NoOptionError, NoSectionError
+from insights.core.plugins import combiner
+from insights.parsers.crio_conf import CrioConf
+
+
+@combiner(CrioConf)
+class AllCrioConf(object):
+    """
+    Combiner for accessing all the crio configuration files. There may be multi
+    files for crio configuration, and the main config file is crio.conf. In the
+    situation that the same section is both in crio.conf and other configuration
+    files, the item in crio.conf has the lowest precedence. This combiner will
+    parse all the CrioConf objects and return a dictionary containing all valid data.
+
+    Sample files::
+
+        /etc/crio/crio.conf:
+
+            [crio]
+            storage_driver = "overlay"
+            storage_option = [
+                    "overlay.override_kernel_check=1",
+            ]
+
+            [crio.runtime]
+            selinux = true
+
+            [crio.network]
+            plugin_dirs = [
+                    "/usr/libexec/cni",
+            ]
+            [crio.metrics]
+
+        /etc/crio/crio.conf.d/00-conmon.conf
+
+            [crio]
+            internal_wipe = true
+            storage_driver = "device mapper"
+            storage_option = [
+                "overlay.override_kernel_check=1",
+            ]
+
+            [crio.api]
+            stream_address = ""
+            stream_port = "10010"
+
+            [crio.runtime]
+            selinux = true
+            conmon = ""
+            conmon_cgroup = "pod"
+            default_env = [
+                "NSS_SDB_USE_CACHE=no",
+            ]
+            log_level = "info"
+            cgroup_manager = "systemd"
+
+    Examples:
+        >>> all_crio_conf.sections()
+        ['crio', 'crio.runtime', 'crio.api', 'crio.network', 'crio.metrics']
+        >>> all_crio_conf.options('crio.api')
+        ['stream_address', 'stream_port']
+        >>> all_crio_conf.files
+        ['/etc/crio/crio.conf', '/etc/crio/crio.conf.d/00-conmon.conf']
+        >>> all_crio_conf.get('crio', 'storage_driver')
+        '"device mapper"'
+
+    Attributes:
+        files (list): The list of configuration file names.
+    """
+    def __init__(self, crio_confs):
+        self.data = {}
+        self.files = []
+        main_data = None
+        conf_d_data = []
+
+        def dict_merge(dest, src):
+            if not src:
+                return
+
+            for section in src.sections():
+                if section not in dest:
+                    dest[section] = {}
+
+                for option in src.items(section):
+                    dest[section][option] = src.get(section, option)
+
+        for crio_conf in crio_confs:
+            self.files.append(crio_conf.file_path)
+            if crio_conf.file_path == "/etc/crio/crio.conf":
+                main_data = crio_conf
+            else:
+                conf_d_data.append(crio_conf)
+
+        dict_merge(self.data, main_data)
+        for crio_conf in conf_d_data:
```

Comment:
I believe you should sort the `confs` got from `crio.conf.d` with the `file_name` at first before the merge.

```
Files in the configuration directory are sorted by name in lexical order and applied in that order. If multiple configuration files specify the same configuration option the setting specified in file sorted last takes precedence over any other value. That is if both 00-default.conf and 10-custom.conf exist in crio.conf.d and both specify different values for a certain configuration option the value from 10-custom.conf will be applied.
```
https://man.archlinux.org/man/community/cri-o/crio.conf.d.5.en

Reply:
Sorted the data with file_name. Please have a look.